src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2770,10 +2770,11 @@ public static <T> T handleVersion(String versionId, DsVersionHandler<T> hdl)
     }
 
     private DatasetVersion getDatasetVersionOrDie(final DataverseRequest req, String versionNumber, final Dataset ds, UriInfo uriInfo, HttpHeaders headers) throws WrappedResponse {
-        return getDatasetVersionOrDie(req, versionNumber, ds, uriInfo, headers, false);
+        //The checkPerms was added to check the permissions ONLY when the dataset is deaccessioned.
+        return getDatasetVersionOrDie(req, versionNumber, ds, uriInfo, headers, false, false);
```

Comment:
I'd suggest making true (the safer option) the default, which would also avoid you having to add true on the methods that you aren't otherwise changing, e.g. line 577.

Reply:
That sounds like a good change. I am just afraid that if I default it to true, someone will use this method and this bug could happen again. 

As an example for `{id}/versions/{versionId}/citation` on line `4023`; it needs to be set to false. What do you think? I sill can create a default signature that doesn't require and defaults it to true, I agree it looks cleaner but still we need the exception in this case. 

I have the change as you suggested ready but wanted to double check. ☕

Reply:
I pushed these changes, probably will add something on the comments just so ppl (if they read the comments 😆 ) know they have to be careful about using the default.

Reply:
I pushed more changes with the comments. I think this should be enough. 😃

Reply:
It's pretty limited in that it only affects deaccessioned versions to begin with, but my thought was to set the default so that someone wouldn't accidentally make info public when it shouldn't be (versus -as you say - make it so that someone might accidentally block a subset of info about a deaccessioned version that should be public).